### PR TITLE
Feature/smar 42

### PR DESCRIPTION
**Pull request by RaznoOleg**
Description changes:
• Connected the frontend to the backend for recovery password;

### General
- [x] Assigned myself to the PR
- [x] Assigned the appropriate labels to the PR
- [x] Assigned the appropriate reviewers to the PR
- [ ] Updated the documentation
- [x] Performed a self-review of my code
- [x] Don't have "any" on my code
- [x] Don't have magic numbers
- [x] Compare only with constants not with strings
- [x] No ternary operator inside ternary operator
- [x] Don't have commented code
- [x] All links in on ENV files
- [x] Used camelCase for variables and functions
- [x] Functions is public only if it's used outside the class
- [x] No hardcoded values
- [x] Check your commit messages meet the [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/).

### Frontend
- [x] Components and business logic are separated
- [x] Colors, Font Size, and Font Name is on the theme or in the constants
- [x] No text in the components, use i18n approach
- [x] No inline styles
- [x] Imports are absolute
- [x] Attach a screenshot if PR has visual changes.

#### Enter the user's email in the forgot password form:
image

#### Confirmation form that opens after clicking button "Submit":
image

#### The link that the user received by the entered email:
image

#### The reset password form that opens after clicking on the link:
image

#### Error if we entered email (in the forgot password form) for which there is no registered user: 
image

#### Changed logo:
image

#### One path for the first and second registration steps:
image
image

#### If the user is already registered in system with the entered email in the fisrt step of registration, an error message appears and user can't continue registration:
image

#### An error message in the second step of registration, if the user entered an invalid phone number:
image

#### Changed field "City" from SelectInput to Input:
image

#### Opening the login form if there is no "/" in the url - base url opens login form:
image

#### Link "Back to Login" and "Go to Login" on the confirmation and activation form:
image
image

### CODE REVIEW

File: src/api/auth/auth.api.ts [hunk 1]
```diff
@@ -18,23 +18,46 @@ export interface AuthSignUpDto {
...
+      .then((res) => res.data);
+  },
+  login(data: AuthLoginDto) {
+    return axios
+      .post<AuthLoginDto>(API_URLS.login, data)
+      .then((res) => res.data);
+  },
+  activation(data: AuthActivationDto) {
+    return axios.get(API_URLS.activation(data.link)).then((res) => res.data);
+  },
```
Comment (lyudad) [inline src/api/auth/auth.api.ts:91]: how you handle errors?

File: src/redux/slices/auth/forgotPassword/index.tsx [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  storage,
+  blacklist: ['isLoading'],
+};
+
+export const forgotPasswordQuery = createAsyncThunk(
+  'forgotPassword/forgotPasswordQuery',
+  async (data: AuthForgotPasswordDto, { rejectWithValue }) => {
+    try {
+      return await authAPI.forgotPassword(data);
+    } catch (err: unknown) {
```
Comment (lyudad) [inline src/redux/slices/auth/forgotPassword/index.tsx]: type is strange

File: src/api/auth/auth.api.ts (hunk 1 above)
Comment (RaznoOleg) [reply src/api/auth/auth.api.ts:91]: fixed

File: src/redux/slices/auth/forgotPassword/index.tsx (hunk 2 above)
Comment (RaznoOleg) [reply src/redux/slices/auth/forgotPassword/index.tsx]: fixed

File: src/api/auth/auth.api.ts [hunk 3]
```diff
@@ -18,23 +18,75 @@ export interface AuthSignUpDto {
...
   signUp(data: AuthSignUpDto) {
     return axios
       .post<AuthSignUpDto>(API_URLS.signUp, data)
-      .then((res) => res.data);
+      .then((res) => res.data)
+      .catch((error) => {
+        throw error.response.data;
+      });
+  },
+  checkEmail(data: AuthCheckEmailDto) {
```
Comment (lyudad) [inline src/api/auth/auth.api.ts]: where is the return data type? have you checked the checklist?

Comment (RaznoOleg) [reply src/api/auth/auth.api.ts]: fixed

**Pull request merged**